arches_for_science/migrations/0005_deploy_triggers_for_primary_descriptor.py

```diff
@@ -0,0 +1,65 @@
+# Generated by Django 4.2.4 on 2023-11-29 16:40
+
+from django.db import migrations
+import pgtrigger.compiler
+import pgtrigger.migrations
+
+
+class Migration(migrations.Migration):
+    dependencies = [
+        ("models", "10260_add_iiifmanifest_globalid"),
+        ("arches_for_science", "0004_improve_primary_descriptor"),
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name="FunctionXGraphProxy",
+            fields=[],
+            options={
+                "proxy": True,
+                "indexes": [],
+                "constraints": [],
+            },
+            bases=("models.functionxgraph",),
+        ),
+        migrations.CreateModel(
+            name="TileModelProxy",
+            fields=[],
+            options={
+                "proxy": True,
+                "indexes": [],
+                "constraints": [],
+            },
+            bases=("models.tilemodel",),
+        ),
+        pgtrigger.migrations.AddTrigger(
+            model_name="functionxgraphproxy",
+            trigger=pgtrigger.compiler.Trigger(
```

Comment:
I would expect the trigger to be removed on the reverse migration, but it persists. When you reverse migrate, are the pgtriggers dropped?

Reply:
Works for me:

Reversing from 0008 to 0004:
```sh
root@d6b16aa0f3ba:/web_root/disco# python3 manage.py migrate arches_for_science 0004
System check identified some issues:

WARNINGS:
?: (urls.W005) URL namespace 'admin' isn't unique. You may not be able to reverse all URLs in this namespace
?: (urls.W005) URL namespace 'oauth2' isn't unique. You may not be able to reverse all URLs in this namespace
Operations to perform:
  Target specific migration: 0004_improve_primary_descriptor, from arches_for_science
Running migrations:
  Rendering model states... DONE
  Unapplying arches_for_science.0008_remove_tilemodelproxy_calculate_multicard_primary_descriptor_single_and_more... OK
  Unapplying arches_for_science.0007_update_primary_description_trigger_handle_static_value... OK
  Unapplying arches_for_science.0006_update_primary_description_trigger_null_handling... OK
  Unapplying arches_for_science.0005_deploy_triggers_for_primary_descriptor... OK
root@d6b16aa0f3ba:/web_root/disco# python3 manage.py dbshell
psql (14.9 (Ubuntu 14.9-1.pgdg22.04+1), server 14.5 (Debian 14.5-1.pgdg110+1))
Type "help" for help.

disco=# \dS tiles
...
Triggers:
    __arches_check_excess_tiles_trigger BEFORE INSERT ON tiles FOR EACH ROW EXECUTE FUNCTION __arches_check_excess_tiles_trigger_function()
    __arches_trg_update_spatial_attributes AFTER INSERT OR DELETE OR UPDATE ON tiles DEFERRABLE INITIALLY DEFERRED FOR EACH ROW EXECUTE FUNCTION __arches_trg_fnc_update_spatial_attributes()
```

I noticed that there was a missing migration on this branch from Friday, that might have been the issue.